src/game/editor/io.cpp

```diff
@@ -398,7 +404,28 @@ int CEditorMap::Load(class IStorage *pStorage, const char *pFileName, int Storag
 							MakeGameLayer(pTiles);
 							MakeGameGroup(pGroup);
 						}
-						else
+						else if(pTilemapItem->m_Flags != 0)  // detect any flagged gametiles
+						{
+							char name_buf[12];
+							IntsToStr(pTilemapItem->m_aName, sizeof(name_buf)/sizeof(int), name_buf);
+							if(str_comp_nocase(name_buf, "Material") == 0)
+							{
+								pTiles = new CLayerMaterial(pTilemapItem->m_Width, pTilemapItem->m_Height);
+								MakeMaterialLayer(pTiles);
+								//MakeGameGroup(pGroup); TODO make a check that this layer is actually in the right group
```

Comment:
@AssassinTee currently nothing is stopping you from hacking a map and adding custom layers and material layers in other groups. Depending on what you want to do this might be usefull, but this is up for discussion and probably never used by anyone usefully. So add a check!